Expose barStyle for NavigatorIOS and TabBarIOS

jacobp100 opened the pull request:
Exposes barStyle property. Code already existed in RCTConvert, so that’s why there’s no conversion code here.

ericvicenti commented:
Nice! Can you add this to the examples?

jacobp100 commented:
I think we’ve usually just showed a single example with a navigation or tab bar using custom text and background colours. It can’t really be used at the same time as custom background colours, so I can either add separate examples for this, or just post screenshots.

ericvicenti commented:
I think separate examples make sense if they cannot be demonstrated together. Ideally you could just toggle the `barStyle` dynamically, but I know that isn't always supported.

jacobp100 commented:
<img width="432" alt="screen shot 2016-11-14 at 21 40 26" src="https://cloud.githubusercontent.com/assets/7275322/20283966/18d40eb4-aab3-11e6-9d49-6d31ed211558.png">
<img width="432" alt="screen shot 2016-11-14 at 21 40 34" src="https://cloud.githubusercontent.com/assets/7275322/20283967/18efba38-aab3-11e6-80fa-6a81362c75ed.png">

mkonicek commented:
Looks good, thanks! I expanded the comments a bit.

@facebook-github-bot shipit

mkonicek commented:
Failed with a Flow error, can you fix it please?

    Libraries/Components/TabBarIOS/TabBarIOS.ios.js line 78 col 30-37: property `barStyle`
    Property not found in
    Libraries/Components/TabBarIOS/TabBarIOS.ios.js line 78 col 19-28: object type

jacobp100 commented:
Just done for TabBarIOS—it seems like there aren’t types for NavigatorIOS.

hramos commented:
Re-triggering tests. We can't land until this is green.

lacker commented:
Again re-triggering tests ;-)

jacobp100 commented:
It was failing on flow. I've just remerged master in. Hopefully this time!

jacobp100 commented:
Should be good now!

lacker commented:
Travis failing on objc-ios, the errors are incomprehensible to me so I am kicking it. https://travis-ci.org/facebook/react-native/builds/199915022

jacobp100 commented:
For the failing bar style, there seems to be an existing comment,

```objc
// TabBarExample on tvOS passes locally but not on Travis
```

I'm not sure why the web socket test is failing. I didn't touch that.

jacobp100 commented:
@lacker We're all good now by the looks of things!

jacobp100 commented:
@lacker Anything I can do my end?

hramos commented:
As far as I can tell, the import failure is unrelated to the code here. We're having issues importing some of the GitHub PRs this week. I'll import this as soon as they're resolved.

ericvicenti commented:
This needs to be rebased before it can merge

jacobp100 commented:
Merged in latest master.

jacobp100 commented:
@hramos

hramos commented:
Ping @ericvicenti who is oncall this week.

jacobp100 commented:
@ericvicenti @hramos Is there anything I can do to help? I can rebase if you want: would you want that on a new branch?

jacobp100 commented:
@hramos @ericvicenti I've updated this to use the new RNTester project, and fixed a flow error. Verified it's still working.

CircleCI seems to be failing on building for Android (looks like some libraries are missing). I've not changed any Android files. Is there anything I need to do?

shergin commented on React/Views/RCTTabBar.h:
```diff
@@ -15,5 +15,8 @@
 @property (nonatomic, strong) UIColor *barTintColor;
 @property (nonatomic, assign) BOOL translucent;
+#if !TARGET_OS_TV
+@property (nonatomic, assign) UIBarStyle *barStyle;
```
Please remove `*` here.

shergin commented on React/Views/RCTTabBar.m:
```diff
@@ -9,6 +9,7 @@
 
 #import "RCTTabBar.h"
 
+#import "RCTConvert.h"
```
Why do we import it `RCTConvert`?

jacobp100 commented:
@shergin Good catches. I've updated the PR.

hramos commented:
After looking at the internal state of this pull request, it seems like we won't be accepting this change. @shergin feel free to re-open if you think otherwise.

jacobp100 commented:
Is there anything I can do to help this?

shergin commented:
Okay, let's try one more time!

shergin commented on RNTester/js/NavigatorIOSBarStyleExample.js:
```diff
@@ -0,0 +1,89 @@
+
+var React = require('react');
+var ReactNative = require('react-native');
+var {
```
Could you please fix code style here?

jacobp100 replied:
Could you clarify what you wanted changing?

shergin replied:
One module - one line, I think you can import all modules directly by names.
